### DISCUSSION TITLE
Redux store not available for children of dynamically imported components

### DISCUSSION THREAD

**Discussion by jfoxworth** [vercel/next.js#31537] [Help]
### What version of Next.js are you using?

9.5.0

### What version of Node.js are you using?

v10.16.0

### What browser are you using?

Chrome, Firefox, Safari

### What operating system are you using?

macOs Mojave

### How are you deploying your application?

Locally deployed with docker server or by running npm run dev.

### Describe the Bug

When a component is imported dynamically, the store is not available to any of its child components. Whenever the useSelector or similar technique is used, an error occurs telling the user that the component must be wrapped in a <Provider> element. This is done at the top level and the store is accessible to all other components at all levels.

The app is not something that external. It's a demo for a well known startup. The data for this demo is refreshed every second and the fact that I cannot read from the store in the lower level components means that I have to do this near the top where the component is dynamically imported. This means that everything beneath that either has to have a class component to limit the refreshes or it will refresh. 

### Expected Behavior

Redux hooks and other techniques to access the store should be available in the children of dynamically imported components.

### To Reproduce

1. Import a component dynamically in your app
2. Create child components for the dynamic import
3. Attempt to access the store with the useSelector method

The _app component ...

`import '../styles/normalize.css';
import '../styles/styles.css';

import PropTypes from 'prop-types';

import Layout from '../components/Layout';
import { wrapper } from '../state/store';

const App = ({ Component, pageProps }) => (
  <Layout>
    <Component {...pageProps} />
  </Layout>
);

App.propTypes = {
  Component: PropTypes.func,
  pageProps: PropTypes.object,
};

export default wrapper.withRedux(App);`

The store ...

`import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction';
import { applyMiddleware, createStore } from 'redux';
import { createWrapper } from 'next-redux-wrapper';

import rootReducer from './rootReducer';
import rootSaga from './rootSaga';

export const makeStore = initialState => {
  const sagaMiddleware = createSagaMiddleware();
  const middleware = applyMiddleware(sagaMiddleware);

  const store = createStore(
    rootReducer,
    initialState,
    composeWithDevTools(middleware),
  );

  sagaMiddleware.run(rootSaga);
  return store;
};

export const wrapper = createWrapper(makeStore);`

The page works fine. Here is the line where the component is dynamically imported into the page.

`import dynamic from 'next/dynamic';
const Venue = dynamic(() => import('../components/Venue'), { ssr: false });
`

Everything else is handled completely normally. The useDispatch and useSelector hooks are imported the same way for every  component. The issue is that they throw an error when any child of the imported component attempts to use them. Not that they work fine within the imported component.

`import { useDispatch, useSelector } from 'react-redux';`

**Comment by cashewnuts** [comment]
I also encounter this issue.

In my case, I found the cause is not nextjs but react-pixi lib, which has limitations of React Context API ([reference](https://github.com/inlet/react-pixi/issues/77#issuecomment-467908120)).

Maybe some of your library have same limitation, I think.
I hope it will help you, @jfoxworth .
